File: Method2/PrimeNumbers.c
#include<stdio.h>

int isPrime(int num) { // Time Complexity O(N)
  if (num < 2) {
    return 0;
  } else {
    int i;
    int half = num / 2;
  
    for (i = 2; i <= half; i++) {
      if (num % i == 0) {
        return 0;
      }
    }
    return 1;
  }
}

int main() {
  int num;

  printf("Enter Value of n... ");
  scanf("%d", &num);

  if (isPrime(num)) {
    printf("%d is a Prime Number.", num);
  } else {
    printf("%d is not a Prime Number.", num);
  }
  return 0;
}

File: Method2/PrimeNumbers.py
def isPrime(num): # Time Complexity O(N)
  if num < 2:
    return False
  else:
    half = num // 2;
    for i in range(2, half + 1):
      if num % i == 0:
        return False
  return True

num = int(input('Enter Value of n... '))
if isPrime(num):
  print(num, 'is a Prime Number.')
else:
  print(num, 'is not a Prime Number.')